spec/models/miq_event_definition_spec.rb

```diff
@@ -27,18 +27,19 @@
       let!(:set1) { create_set!('host_operations') }
       let!(:set2) { create_set!('evm_operations') }
 
-      context 'and CSV file with an event definition linked to one of the sets' do
-        let(:csv) do
-          <<-CSV.strip_heredoc
-            name,description,event_type,set_type
-            #
-            evm_server_start,EVM Server Start,Default,host_operations
-          CSV
+      context 'and YML file with an event definition linked to one of the sets' do
+        let(:yml) do
+          <<-YML.strip_heredoc
+            - name: evm_server_start
+              description: EVM Server Start
+              event_type: Default
+              set_type: host_operations
+          YML
         end
 
-        context 'seeding default event definitions from that csv' do
+        context 'seeding default event definitions from that yml' do
           before do
-            allow(File).to receive(:open).and_return(StringIO.new(csv))
+            allow(File).to receive(:open).and_return(StringIO.new(yml))
```

Comment:
This may not be right because I don't think YAML uses File.open under the covers.  This might be better as
```suggestion
            allow(YAML).to receive(:load_file).and_return(yml)
```

And then up above in the `let(:yml)`, instead of giving raw yaml, you can just put a normal Array with a Hash in it

```ruby
            let(:yml) do
              [{
                "name"        => "evm_server_start",
                "description" => "EVM Server Start",
                "event_type"  => "Default",
                "set_type"    => "host_operations"
              }]
            end
```

@jrafanie what do you think on this particular test?

Reply:
I think the part of the code in event definition that reads the file and yaml loads it should be a method, so you can replace it in the tests:

```ruby
fname = File.join(FIXTURE_DIR, "miq_event_definition_events.yml")
YAML.load_file(fname).each do |event|
```

Could become:

```ruby
def event_definitions_from_path(path)
  YAML.load_file(path)
end

fname = File.join(FIXTURE_DIR, "miq_event_definition_events.yml")
event_definitions_from_path(fname).each do |event|
```

similar in event definition set:

```ruby
    YAML.load_file(fixture_path).each do |set|
```

You could then do `allow(MiqEventDefinition).to receive(:event_definitions_from_path).and_return(my_test_hash)`.

Reply:
Hey @jrafanie, I tried to implement the changes you suggested but for whatever reason when running the test cases the `event_definitions_from_path` function isn't detected

```
Failure/Error:
           event_definitions_from_path(fname).each do |event|
             set_type = event.delete('set_type')
       
             rec = event_defs[event['name']].try(:first)
             if rec.nil?
               _log.info("Creating [#{event['name']}]")
               rec = create(event)
               (event_defs[event['name']] ||= []) << rec
             else
               rec.attributes = event
     
     NoMethodError:
       undefined method `event_definitions_from_path' for #<Class:0x00007f89889c1288>
```

Reply:
sorry, it's a class method...

```ruby
def self.event_definitions_from_path(path)
  YAML.load_file(path)
end
```